common/src/main/java/io/netty/util/ResourceLeakDetectorFactory.java

```diff
@@ -88,7 +88,7 @@ public static void setResourceLeakDetectorFactory(ResourceLeakDetectorFactory fa
      */
     @SuppressWarnings("deprecation")
     public <T> ResourceLeakDetector<T> newResourceLeakDetector(Class<T> resource, int samplingInterval) {
-        return newResourceLeakDetector(resource, ResourceLeakDetector.SAMPLING_INTERVAL, Long.MAX_VALUE);
```

Comment:
What about to add a check `ObjectUtil.checkPositive(samplingInterval, "samplingInterval")` ?

Reply:
Yes, this is feasible. It is possible that this parameter is the user's configuration. Is it appropriate to make an assertion here?@normanmaurer